tests/MongoDB.Driver.Core.Tests/Core/Authentication/ScramSha1AuthenticatorTests.cs

```diff
@@ -189,27 +205,47 @@ public void Constructor_should_throw_an_ArgumentNullException_when_credential_is
                 actualRequestIds[i].Should().BeInRange(expectedRequestId + i, expectedRequestId + 10 + i);
             }
 
-            sentMessages[0].Should().Be(
-                $"{{opcode: \"query\", " +
+            var saslStartMessage = BsonDocument.Parse(
+                @"{ opcode: ""query""," +
                 $"  requestId: {actualRequestIds[0]}, " +
-                $"  database: \"source\", collection: \"$cmd\", batchSize: -1, slaveOk: true, " +
-                $"  query: {{saslStart: 1, mechanism: \"SCRAM-SHA-1\", " +
+                @"  database: ""source"",
+                    collection: ""$cmd"",
+                    batchSize: -1,
+                    slaveOk: true,
+                    query: { saslStart: 1,
+                             mechanism: ""SCRAM-SHA-1""," +
                 $"           payload: new BinData(0, \"biwsbj11c2VyLHI9ZnlrbytkMmxiYkZnT05Sdjlxa3hkYXdM\")" +
                 @"           options: { skipEmptyExchange: true }}}");
-            sentMessages[1].Should().Be("{opcode: \"query\", requestId: " + actualRequestIds[1] + ", database: \"source\", collection: \"$cmd\", batchSize: -1, slaveOk: true, query: {saslContinue: 1, conversationId: 1, payload: new BinData(0, \"Yz1iaXdzLHI9ZnlrbytkMmxiYkZnT05Sdjlxa3hkYXdMSG8rVmdrN3F2VU9LVXd1V0xJV2c0bC85U3JhR01IRUUscD1NQzJUOEJ2Ym1XUmNrRHc4b1dsNUlWZ2h3Q1k9\")}}");
-            if (useLongAuthentication)
-            {
-                sentMessages[2].Should().Be(
+
+            var saslContinueMessage = BsonDocument.Parse(
+                @"{ opcode: ""query""," +
+                $"  requestId: {(useSpeculativeAuthenticate ? actualRequestIds[0] : actualRequestIds[1])}," +
+                @"  database: ""source"",
+                    collection: ""$cmd"",
+                    batchSize: -1,
+                    slaveOk: true,
+                    query: { saslContinue: 1,
+                             conversationId: 1, " +
+                $"           payload: new BinData(0, \"Yz1iaXdzLHI9ZnlrbytkMmxiYkZnT05Sdjlxa3hkYXdMSG8rVmdrN3F2VU9LVXd1V0xJV2c0bC85U3JhR01IRUUscD1NQzJUOEJ2Ym1XUmNrRHc4b1dsNUlWZ2h3Q1k9\")}}}}");
+             var saslOptionalFinalMessage = new Lazy<BsonDocument>(() => BsonDocument.Parse(
```

Comment:
This needs to be lazy else we get index out of bounds exceptions when using short form authentication i.e. `useLongAuthentication` is `false`

Reply:
Why create this variable here if we don't yet know if we need it? See below.

Reply:
I've taken the refactor!